protobuf/buf.gen.go-server.yaml

```diff
@@ -3,16 +3,16 @@ managed:
   enabled: true
 plugins:
   - remote: buf.build/protocolbuffers/plugins/go:v1.28.0-1
-    out: ./proto/go-server
+    out: ../proto/go-server
```

Comment:
What impact does this change have?

Reply:
I noticed that the [buf.gen.go-server.yaml](https://github.com/open-feature/schemas/pull/15/files/6c01bb297f8d1e894ec954dce5d757b59a7602e8#diff-349b668a9ee6f43e5758df546881078238fd44ebf71e7497cdd21c59dd05732f) output path differs from the other buf.gen.*.yaml files, seeing as these changes to the schema will require updates to flagd I felt it would be a good time to resolve the issue